<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CSettingsDialog</class>
 <widget class="QDialog" name="CSettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>840</width>
    <height>368</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CSV saving parameters</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_7">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>CSV files saving</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>&amp;Directory for CSV files</string>
              </property>
              <property name="buddy">
               <cstring>editCSVDir</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="editCSVDir"/>
            </item>
            <item>
             <widget class="QToolButton" name="btnCSVDir">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>&amp;File name template</string>
              </property>
              <property name="buddy">
               <cstring>editCSVTemplate</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="editCSVTemplate"/>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="checkRestoreCSV">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This settings only for interactive mode.&lt;/p&gt;&lt;p&gt;When starting with '-start' key from command promnt, this option is always activated.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Restore CSV recording after connection failure</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Signal plot</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Plot &amp;vertical size</string>
              </property>
              <property name="buddy">
               <cstring>spinPlotVerticalSize</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinPlotVerticalSize">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="suffix">
               <string> pixels</string>
              </property>
              <property name="minimum">
               <number>30</number>
              </property>
              <property name="maximum">
               <number>4000</number>
              </property>
              <property name="value">
               <number>100</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="checkPlotShotScatter">
            <property name="text">
             <string>Show scatter points</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkAntialiasing">
            <property name="text">
             <string>Antialiasing</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Connection</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QGridLayout" name="gridLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>&amp;Max error count before connection stop</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
              <property name="buddy">
               <cstring>spinMaxRecErrorCount</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinMaxRecErrorCount">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Maximum error count for daveReadBytes function. &lt;/p&gt;&lt;p&gt;This counter increases in each PLC cycle on connection failures, inaccessible or incorrect addresses of PLC memory areas, specified in variables.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="maximum">
               <number>99999</number>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>(0 - don't stop)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>TCP &amp;connect retries when starting connection</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
              <property name="buddy">
               <cstring>spinConnectRetryCount</cstring>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="spinConnectRetryCount">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;TCP connection retries when starting online mode.&lt;/p&gt;&lt;p&gt;Do not set &lt;span style=&quot; font-weight:600;&quot;&gt;this parameter&lt;/span&gt; and &lt;span style=&quot; font-weight:600;&quot;&gt;reconnection retries count&lt;/span&gt; both equal to zero.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="maximum">
               <number>99999</number>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>(0 - infinite)</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Connection &amp;timeout </string>
              </property>
              <property name="buddy">
               <cstring>spinNetTimeout</cstring>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="spinNetTimeout">
              <property name="suffix">
               <string> ms</string>
              </property>
              <property name="minimum">
               <number>1000</number>
              </property>
              <property name="maximum">
               <number>120000</number>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>&amp;Pause before connection retry</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
              <property name="buddy">
               <cstring>spinWaitBeforeRetryConnect</cstring>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSpinBox" name="spinWaitBeforeRetryConnect">
              <property name="suffix">
               <string> sec</string>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>99999</number>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Failed connection &amp;retry count</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
              <property name="buddy">
               <cstring>spinTotalRetryCount</cstring>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QSpinBox" name="spinTotalRetryCount"/>
            </item>
            <item row="4" column="2">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>(0 - infinite)</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Interface</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QCheckBox" name="checkSuppressMsgBoxes">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Suppress message boxes with error and warning messages. All messages saving to log window.&lt;/p&gt;&lt;p&gt;Very useful and memory conservative on long background works when starting with '-start' key from command prompt.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Do not show interactive message boxes</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>OK</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_2">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>editCSVDir</tabstop>
  <tabstop>btnCSVDir</tabstop>
  <tabstop>editCSVTemplate</tabstop>
  <tabstop>checkRestoreCSV</tabstop>
  <tabstop>spinPlotVerticalSize</tabstop>
  <tabstop>checkPlotShotScatter</tabstop>
  <tabstop>checkAntialiasing</tabstop>
  <tabstop>spinMaxRecErrorCount</tabstop>
  <tabstop>spinConnectRetryCount</tabstop>
  <tabstop>spinNetTimeout</tabstop>
  <tabstop>spinWaitBeforeRetryConnect</tabstop>
  <tabstop>spinTotalRetryCount</tabstop>
  <tabstop>checkSuppressMsgBoxes</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>pushButton_2</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>CSettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>165</x>
     <y>114</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>CSettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>419</x>
     <y>130</y>
    </hint>
    <hint type="destinationlabel">
     <x>77</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
